File: samples/wasm/c/calculator/calculator.c
#include <emscripten.h>

/* EMSCRIPTEN_KEEPALIVE adds a function to the exports in the wasm file */

EMSCRIPTEN_KEEPALIVE
int add(int x, int y) {
  return x + y;
}

EMSCRIPTEN_KEEPALIVE
int subtract(int x, int y) {
  return x - y;
}

File: samples/wasm/go/calculator/calculator.go
//go:build wasm && js && !wasi

package main

// Attempt at dirty hack to avoid TinyGo depending upon fd_write
// See https://github.com/tinygo-org/tinygo/issues/1123
// But in TinyGo 0.28.1 this no longer prevents the import from being generated
//
// export fd_write
//func fd_write() (errno uint) {
//	return 0
//}

//go:export add
func add(x int32, y int32) int32 {
	return x + y
}

//go:export subtract
func subtract(x int32, y int32) int32 {
	return x - y
}

// Needed even though it is a noop
func main() {}
